### PR TITLE
Add option to restrict new version to certain groups

### PR DESCRIPTION
**Pull request by woodwardmw**

### CODE REVIEW

File: bible_routes/v3/version_routes.py [hunk 1]
```diff
@@ -82,6 +84,8 @@ async def add_version(
     user_group_ids = [group_id[0] for group_id in user_group_ids_query.all()]
 
     for group_id in user_group_ids:
+        if add_to_groups and group_id not in add_to_groups:
```
Comment (paquinteroc) [inline bible_routes/v3/version_routes.py:87]: need to check if add to groups is None

File: bible_routes/v3/version_routes.py [hunk 2]
```diff
@@ -82,6 +84,8 @@ async def add_version(
...
+            continue
         access = BibleVersionAccess(bible_version_id=new_version.id, group_id=group_id)
         db.add(access)
     db.commit()
```
Comment (paquinteroc) [inline bible_routes/v3/version_routes.py:91]: will need a test case for this payload

File: bible_routes/v3/version_routes.py (hunk 1 above)
Comment (woodwardmw) [reply bible_routes/v3/version_routes.py:87]: I think `if add_to_groups` here should be the check of whether it's `None`. So if it's `None`, you don't skip any group_ids. But if it's a list, you skip those that are not in the list.

File: bible_routes/v3/version_routes.py (hunk 2 above)
Comment (woodwardmw) [reply bible_routes/v3/version_routes.py:91]: Ok yep, I'll add a test to this PR.

File: bible_routes/v3/version_routes.py [hunk 3]
```diff
@@ -82,6 +84,8 @@ async def add_version(
     user_group_ids = [group_id[0] for group_id in user_group_ids_query.all()]
```
Comment (paquinteroc) [inline bible_routes/v3/version_routes.py:84]: should we return a 422 if one of the groups in the list is not an actual group in the db? I do not have strong feelings either way but it maybe useful

File: test/test_bible_routes/test_version_routes.py [hunk 4]
```diff
@@ -98,38 +98,119 @@ def test_regular_user_create_list_and_delete_version(
...
+            .first()
+        )
+        assert user_group is not None
+        user_group = (
+            db_session.query(UserGroup)
+            .filter(UserGroup.user_id == user_2.id, UserGroup.group_id == group_3.id)
+            .first()
+        )
+        assert user_group is not None
+
```
Comment (paquinteroc) [inline test/test_bible_routes/test_version_routes.py:149]: i like the db setup here, i may copy for future reference, with this flow and the assertions the developer never has to look back at the conftest file

**Review by paquinteroc — APPROVED** [review]

File: test/test_bible_routes/test_version_routes.py [hunk 5]
```diff
@@ -98,38 +98,119 @@ def test_regular_user_create_list_and_delete_version(
...
+        assert versions[0]["id"] == version_id_1
+        assert versions[1]["id"] == version_id_2
+
+        # Verify user 2 can access only the first version
+        headers = {"Authorization": f"Bearer {regular_token2}"}
+        list_response = client.get(f"{prefix}/version", headers=headers)
+        assert list_response.status_code == 200
+        versions = list_response.json()
+        assert len(versions) == 1, "User 2 should only see one version"
+        assert versions[0]["id"] == version_id_1
```
[the remark this answers is not in the document]
Comment (woodwardmw) [reply test/test_bible_routes/test_version_routes.py:194]: OK yep, I'll add that.

File: bible_routes/v3/version_routes.py (hunk 3 above)
Comment (woodwardmw) [reply bible_routes/v3/version_routes.py:84]: Yeah, I could go either way... let's maybe leave it for now.

**Pull request merged**